src/store/modules/segmentationModule.js

```diff
@@ -0,0 +1,1411 @@
+  fillAlphaInactive: 0.1,
+  outlineAlpha: 0.7,
+  outlineAlphaInactive: 0.35,
+  outlineWidth: 3,
+};
+
+/**
+ * A map of `firstImageId` to associated `BrushStackState`, where
+ * `firstImageId` is the `imageId` of the first image in a stack.
+ *
+ * @typedef {Object} Series
+ */
+
+/**
+ * @typedef {Object} BrushStackState An object defining a set of 3D labelmaps
+ *    associated with a specific cornerstone stack.
+ * @property {string} activeLabelmapIndex The index of the active `Labelmap3D`.
+ * @property {Labelmap3D[]} labelmaps3D An array of `Labelmap3D` objects.
+ */
+
+/**
+ * An
+ *
+ * @typedef {Object} Labelmap3D An object defining a 3D labelmap.
+ * @property {ArrayBuffer}  buffer An array buffer to store the pixel data of the `Labelmap3D` (2 bytes/voxel).
+ * @property {Labelmap2D[]} labelmaps2D array of `labelmap2D` views on the `buffer`, indexed by in-stack
+ *                          image positions.
+ * @property {Object[]} metadata An array of metadata per segment. Metadata is optional and its form is
+ *                               application specific.
+ * @property {number} activeSegmentIndex The index of the active segment for this `Labelmap3D`.
+ * @property {boolean[]} segmentsVisible The visibility of segments on this labelmap.
+ *                                       If an element is undefined, the visibility of that defaults to true.
+ */
+
+/**
+ * @typedef {Object} Labelmap2D An object defining a 2D view on a section of a `Labelmap3D`'s `buffer`.
+ * @property {Uint16Array} pixelData A 2D view on a section of the parent `Labelmap3D`'s `buffer`.
+ * @property {number[]} segmentsOnLabelmap An array of segments present in the `pixelData`.
+ * @property {boolean} invalidated Whether the data has been changed recently, and a new `ImageBitmap` should be created
+ *                                 for the segmentation fill should this `labelmap2D` be rendered.
+ */
+
+/**
+ * GetMetadata - Returns the metadata object for a partiular segment if
+ * segmentIndex is specified, otherwise returns an array of all segment metadata
+ * for the labelmap.
+ *
+ * @param  {HTMLElement|string} elementOrEnabledElementUID   The cornerstone enabled
+ *                                                    element or its UUID.
+ * @param  {number} [labelmapIndex]    If undefined, defaults to the active
+ *                                     labelmap index.
+ * @param  {number} [segmentIndex]     The segment index.
+ * @returns {Object|Object[]}          A metadata object or an array of
+ *                                     metadata objects.
+ */
+function getMetadata(elementOrEnabledElementUID, labelmapIndex, segmentIndex) {
+  const element = _getEnabledElement(elementOrEnabledElementUID);
+
+  if (!element) {
+    return;
+  }
+
+  const stackState = getToolState(element, 'stack');
+  const stackData = stackState.data[0];
+  const firstImageId = stackData.imageIds[0];
+
+  const brushStackState = state.series[firstImageId];
```

Comment:
Hmm.. What if we have two stacks that have the same first image id?
Is that a thing? Hanging protocol issue potentially?

Reply:
Yeah this could be an issue, we've discussed before that this isn't perfect. Cornerstone has no `stackUUID` or the like, perhaps we should add this? We can discuss today.